### PR TITLE
agile: add new package

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/agile/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/agile/package.py]: What does the question mark mean here:
```python
# only possible with python2 (?)
```
?

File: var/spack/repos/builtin/packages/agile/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/agile/package.py:32]: I think hard-coding optimization flags in the recipe is not considered a best practice. Is there any way to avoid it?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/agile/package.py
Comment (vvolkl) [reply var/spack/repos/builtin/packages/agile/package.py]: Supposed to mean "probably" / "not extensively tested". Will update the comment to be clearer.

File: var/spack/repos/builtin/packages/agile/package.py
Comment (vvolkl) [reply var/spack/repos/builtin/packages/agile/package.py:32]: I tend to agree, but this recipe is mostly for archiving purposes, documenting how this package was built. It might be needed as a dependency for another generator, but mostly we just want to reproduce how it is currently built, and that happens to be the way. If there's another way to set this as a default but leave it configurable, I'm happy to do it, but I think it's ok in this case.

File: var/spack/repos/builtin/packages/agile/package.py
Comment (vvolkl) [inline var/spack/repos/builtin/packages/agile/package.py]:
```suggestion
    depends_on('python')
    depends_on('swig')
    depends_on('py-future')
```
Oh, these also aren't necessary when  disabling pyext, will remove.

File: var/spack/repos/builtin/packages/agile/package.py
Comment (alalazo) [reply var/spack/repos/builtin/packages/agile/package.py:32]: Fine with me if you add yourself as a maintainer :slightly_smiling_face: (so it's clear who to ping in case of questions, if any)

File: var/spack/repos/builtin/packages/agile/package.py
Comment (vvolkl) [inline var/spack/repos/builtin/packages/agile/package.py:17]:
```suggestion

     maintainers = ['vvolkl']
 
```

File: var/spack/repos/builtin/packages/agile/package.py
Comment (vvolkl) [reply var/spack/repos/builtin/packages/agile/package.py:32]: Sure!

**Review by alalazo — APPROVED** [review]

**Review by alalazo — CHANGES_REQUESTED** [review]
LGTM, once tests are :heavy_check_mark:

**Comment by alalazo** [comment]
@vvolkl

**Comment by vvolkl** [comment]
@alalazo Seems like the test failures are unrelated to the PR:
```
==> Modified files
  var/spack/repos/builtin/packages/agile/package.py
==> Error: cannot bootstrap any of the isort executables from spec "py-isort@4.3.5: %gcc target=x86_64"
==> Bootstrapping py-isort from pre-built binaries
==> Bootstrapping clingo from pre-built binaries
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.2/build_cache/linux-centos7-x86_64/gcc-10.2.1/clingo-bootstrap-spack/linux-centos7-x86_64-gcc-10.2.1-clingo-bootstrap-spack-2l45t4kw3cqqwj6vbxhfwhzlo6b3q2p4.spack
==> Installing "clingo-bootstrap@spack%gcc@10.2.1~docs~ipo+python+static_libstdcpp build_type=Release arch=linux-centos7-x86_64" from a buildcache
Error: Process completed with exit code 3.
```
Can you try restarting the tests?

**Comment by alalazo** [comment]
@vvolkl These usually are symptoms of some syntax error or similar, let me double check.

File: var/spack/repos/builtin/packages/agile/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/agile/package.py]:
```suggestion
    maintainers = ['vvolkl']
```

**Review by alalazo — CHANGES_REQUESTED** [review]

**Comment by vvolkl** [comment]
Ah right, good catch, thanks.

**Review by alalazo — DISMISSED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**
